src/lockscreen.js

```diff
@@ -56,24 +59,42 @@ var Lockscreen = GObject.registerClass({
     set key(key) {
         if ('_key' in this && this._key === key)
             return;
-        if (this._managerChangedId !== 0)
-            this._manager.disconnect(this._managerChangedId);
-        this._managerChangedId = this._manager.connect(
-            `changed::${key}`, this._onChanged.bind(this));
+        if (this._keyChangedId !== 0)
+            this._manager.disconnect(this._keyChangedId);
+        this._keyChangedId = this._manager.connect(
+            `changed::${key}`, this._updateLockStateWithKey.bind(this));
         this._key = key;
-        this._updateLockState();
+        this._updateLockStateWithKey();
     }
 
-    _onChanged() {
-        this._updateLockState();
+    get lock() {
+        return this._lock;
+    }
 
-        /* TODO check for key and display can-be-opened graphics */
+    set lock(lock) {
+        if ('_lock' in this && this._lock === lock)
+            return;
+        if (this._lockChangedId !== 0)
+            this._manager.disconnect(this._lockChangedId);
+        this._lockChangedId = this._manager.connect(
+            `changed::${lock}`, this._updateLockStateWithLock.bind(this));
+        this._lock = lock;
+        this._updateLockStateWithLock();
     }
 
-    _updateLockState() {
+    _updateLockStateWithKey() {
         if (!this._key)
             return;
-        if (!this._manager.isUnlocked(this._key))
+        if (this._manager.hasKey(this._key)) {
+
+            /* TODO update the lock-screen graphics with can-be-opened image */
```

Comment:
It would be great to add at least a mock graphic for now, for the quests walkthrough.